src/odemis/driver/test/orsay_test.py

```diff
@@ -2026,132 +2080,265 @@ def setUpClass(cls):
+            self.assertIsInstance(image, model.DataArray)
+            self.assertEqual(image.dtype, numpy.uint16)
+            self.assertEqual(image.shape, self.scanner.resolution.value)
+            self.assertIn(model.MD_ACQ_DATE, image.metadata)
+            stop_t = time.time()
+            logging.info("Expected time: %s s, actual time: %g s", scan_time, stop_t - start_t)
+            durations.append(stop_t - start_t)
 
-        cls.dataflow = cls.detector.data
-        cls._init_dwellTime = cls.fibbeam.dwellTime.value
-        cls.fibbeam.dwellTime.value = min(cls.fibbeam.dwellTime.choices)  # Speeds up the test cases
+        logging.info("Durations: %s", durations)
 
-    @classmethod
-    def tearDownClass(cls):
-        cls.fibbeam.dwellTime.value = cls._init_dwellTime
+    def test_get_not_asap(self):
+        scan_time = numpy.prod(self.scanner.resolution.value) * self.scanner.dwellTime.value
+        durations = []
 
-    def test_get(self):
-        self.scanner.blanker.value = False
-        image = self.dataflow.get(asap=False)
-        self.assertIsInstance(image, model.DataArray)
-        self.assertEqual(image.dtype, numpy.uint16)
-        self.assertEqual(image.shape, self.fibbeam.imageFormat.value)
+        self.size = self.scanner.resolution.value
+        self.left = 50
+        self.detector.data.subscribe(self.receive_image)
 
-        # Test with asap = True
-        image = self.dataflow.get(asap=True)
-        self.assertIsInstance(image, model.DataArray)
-        self.assertEqual(image.dtype, numpy.uint16)
-        self.assertEqual(image.shape, self.fibbeam.imageFormat.value)
-        self.scanner.blanker.value = True
+        try:
+            # Test with asap = False (should not have any difference when the stream is not already playing)
+            for i in range(5):
+                time.sleep(0.5)  # Pretend to do something long and useful
+                start_t = time.time()
+                image = self.detector.data.get(asap=False)
+                self.assertIsInstance(image, model.DataArray)
+                self.assertEqual(image.dtype, numpy.uint16)
+                self.assertEqual(image.shape, self.scanner.resolution.value)
+                self.assertGreaterEqual(image.metadata[model.MD_ACQ_DATE], start_t)
+                stop_t = time.time()
+                logging.info("Expected time: %s s, actual time: %g s", scan_time, stop_t - start_t)
+                durations.append(stop_t - start_t)
+        finally:
+            self.detector.data.unsubscribe(self.receive_image)
+
+        logging.info("Durations: %s", durations)
+
+    # TODO: test cropping (while acquiring)
 
     def test_get_varying_dwell_time(self):
-        self.skipTest(f"By default this test is skipped. It test the functionality on changing dwell times. User input is required")
+        # self.skipTest(f"By default this test is skipped. It test the functionality on changing dwell times. User input is required")
```

Comment:
Probably this comment can go away? (or if changing dwelltimes is not that stable yet it should remain uncommented?)

Reply:
Hi Kornee, nice to have you shinning up here!
The change of dwell time is now stable (as far as I could test). So indeed this line can completely go away.